nvbench/device_info.cuh

```diff
@@ -175,6 +208,30 @@ struct device_info
   /// @return True if ECC is enabled on this device.
   [[nodiscard]] bool get_ecc_state() const { return m_prop.ECCEnabled; }
 
+  /// @return True if CUPTI supports this device.
+  [[nodiscard]] bool is_cupti_supported() const
+  {
+#ifdef NVBENCH_HAS_CUPTI
+    return m_prop.major >= 7;
+#else
+    return false;
+#endif
+  }
+
+#ifdef NVBENCH_HAS_CUPTI
+  [[nodiscard]] CUcontext get_context() const
+  {
+    if (!is_active())
+    {
+      throw std::runtime_error("get_context is called for inactive device");
```

Comment:
Move this function into `device_info.cu` (to avoid leaking fmt) and use the NVBENCH_THROW macro to embed file/line information.